src/adapter/sources.ts

```diff
@@ -320,6 +341,18 @@ export class SourceContainer {
     return undefined;
   }
 
+  sourceByReference(refNum: number): Source | undefined {
```

Comment:
It seems like the new sourceByReference and sourceByUrl are doing basically the same thing as source(), you can pass in a fake Dap.Source or change it to take a new type, but I don't think we need these new getters and the new sourceByUrl map when we have sourceByAbsolutePath.

Reply:
Yea, elsewhere we just create partial dap sources

Reply:
Oh ok, I see. Thanks!